File: Msys_Programs/MSys_Projects/LeetCode/GoLang/Search_insert_Position.go
package main

import "fmt"

func searchInsert(nums []int, target int) int {
	if nums[len(nums)-1] < target {
		return len(nums)
	}
	for i := 0; i < len(nums); i++ {
		if nums[i] == target {
			return i
		}
		if nums[i] > target {
			return i
		}
	}
	return -1
}

func main() {
	O := []int{1, 3, 5, 6}
	t := 2
	fmt.Println(searchInsert(O, t))
}

File: Msys_Programs/MSys_Projects/LeetCode/Python/Search_insert_Position.py
class Solution:
    def searchInsert(self, nums, target):
        idx = 0
        for i in range(len(nums)):
            if nums[i] == target:
                return i
            if nums[i] < target:
                idx = i + 1
            else:
                break
                
        return idx

a = [1,3,5,6]
t = 5
O = Solution()
print(O.searchInsert(a, t))